### PR TITLE
ARROW-15890: [CI][Python] Use venv instead of virtualenv

### PR DESCRIPTION
**Pull request by pitrou**

### CODE REVIEW

**Comment by pitrou** [comment]
@github-actions crossbow submit wheel-macos*

File: python/examples/minimal_build/README.md [hunk 1]
```diff
@@ -24,32 +24,32 @@ an eye toward maximum portability across different Linux distributions. This
 may help for contributors debugging build issues caused by their local
 environments.
 
-## Fedora 31
+## Fedora 35
```
Comment (pitrou) [inline python/examples/minimal_build/README.md:27]: Note: we should perhaps test those minimal builds as part of our nightly builds, but that's outside the scope of this PR.

**Comment by kszucs** [comment]
@github-actions crossbow submit wheel-macos-big-sur-cp39-universal2

**Comment by kszucs** [comment]
@github-actions crossbow submit wheel-macos-big-sur-cp39-universal2

**Comment by kszucs** [comment]
@github-actions crossbow wheel-macos-big-sur-cp310-universal2

**Comment by kszucs** [comment]
@github-actions crossbow submit wheel-macos-big-sur-cp310-universal2

**Review by kszucs — APPROVED** [review]
Thanks Antoine!

**Comment by kszucs** [comment]
The remaining failures are due to a failing test case `test_debug_memory_pool_disabled` which is unrelated to this PR.

**Pull request closed without merging**